### ISSUE TITLE
jupyterlab install nbwidgets dont work

### ISSUE THREAD

**Issue by Mblp4ur** [jupyter-widgets/ipywidgets#1435]
`jupyter labextension install @jupyterlab/nbwidgets`
output:

    npm pack @jupyterlab/nbwidgets
    Traceback (most recent call last):
    File "/usr/local/bin/jupyter-labextension", line 11, in
    sys.exit(main())
    File "/usr/local/lib/python2.7/dist-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, kwargs)
    File "/usr/local/lib/python2.7/dist-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/labextensions.py", line 167, in start
    super(LabExtensionApp, self).start()
    File "/usr/local/lib/python2.7/dist-packages/jupyter_core/application.py", line 256, in start
    self.subapp.start()
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/labextensions.py", line 57, in start
    for arg in self.extra_args]
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/commands.py", line 110, in install_extension
    errors = _validate_compatibility(extension, deps, core_data)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/commands.py", line 489, in _validate_compatibility
    overlap = _test_overlap(core_deps[key], value)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/commands.py", line 510, in _test_overlap
    r1 = Range(spec1, True)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 669, in init
    xs = [self.parse_range(r.strip()) for r in re.split(r"\s||\s", range_)]
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 723, in parse_range
    set_ = [make_comparator(comp, loose) for comp in set_]
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 593, in comparator
    return Comparator(comp, loose)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 605, in init
    self.parse(comp)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 628, in parse
    self.semver = semver(m.group(2), self.loose)
    File "/usr/local/lib/python2.7/dist-packages/jupyterlab/semver.py", line 315, in semver
    raise ValueError("Invalid Version: {}".format(version))
    ValueError: Invalid Version: 0.7.0-0```

```git clone https://github.com/jupyter-widgets/ipywidgets.git
cd ipywidgets/jupyterlab_widgets
npm install
npm run build
jupyter labextension link .```

also

**Comment by Mblp4ur** [comment]
```
jupyter lab --version
0.24.1
jupyter --version
4.3.0

```

**Comment by jasongrout** [comment]
I've reproduced this...

**Comment by jasongrout** [comment]
This appears to be a python 2 incompatibility with the jupyter semver module. I've opened https://github.com/jupyterlab/jupyterlab/pull/2486 - you can make that change manually in jupyterlab's semver.py file and the installation should proceed. Upgrading to python 3 would also do the trick.

**Comment by jasongrout** [comment]
Please be aware that we are discussing only supporting python 3 in JupyterLab: https://github.com/jupyterlab/jupyterlab/issues/2096

**Comment by jasongrout** [comment]
Closing - as the issue should be resolved in jupyterlab/jupyterlab#2486

**Comment by jasongrout** [comment]
Thanks for flagging this!
